### PR TITLE
test: pod-store & pod-sync testing

### PR DESCRIPTION
**Pull request by jonasbovyn**

### CODE REVIEW

File: packages/semcom-node/lib/component/services/component-transformer.service.spec.ts [hunk 1]
```diff
@@ -196,10 +196,13 @@ describe('ComponentTransformerService', () => {
 
     });
 
-    it('should throw if at least one metadata throws an error', () => {
+    it("should return only the metadata's that were correctly parsed", () => {
```
Comment (AbelVandenBriel) [inline packages/semcom-node/lib/component/services/component-transformer.service.spec.ts:199]:
```suggestion
    it("should return only the metadata that was correctly parsed", () => {
```

File: packages/semcom-node/lib/store/services/component-pod-store.service.spec.ts [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+  interface ComponentMetadataMock {
+    component: ComponentMetadata;
+    filename: string;
+  }
+
+  const streamToBuffer = (stream: NodeJS.ReadableStream): Promise<string> => new Promise((resolve) => {
+
+    const bufs = [];
+    stream.on('data', (chunk) => bufs.push(chunk));
+    stream.on('end', () => resolve(bufs.join('')));
```
Comment (AbelVandenBriel) [inline packages/semcom-node/lib/store/services/component-pod-store.service.spec.ts:34]:
```suggestion
    stream.on('end', () => resolve(Buffer.concat(buf)));
```
Buffer has it's own method to concatenate an array of Buffers

File: packages/semcom-node/lib/store/services/component-pod-store.service.spec.ts [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+
+  });
+
+  const mockComponents = (components: ComponentMetadataMock[]) => {
+
+    fetchMock = (fetch as jest.MockedFunction<typeof fetch>);
```
Comment (AbelVandenBriel) [inline packages/semcom-node/lib/store/services/component-pod-store.service.spec.ts:40]: Can't we just use the library fetchMock? https://www.npmjs.com/package/fetch-mock

We've used this before, you can look for some examples in the id-broker

File: packages/semcom-node/lib/store/services/component-pod-store.service.spec.ts [hunk 4]
```diff
@@ -0,0 +1,215 @@
...
+          label: 'test2',
+          description: 'defg',
+          author: '',
+          tag: 'test2',
+          version: '0.123',
+          latest: true,
+          shapes: [ 'http://digita.ai/voc/test#test2' ],
+        },
+        filename: 'test2.ttl',
+      } ];
```
Comment (AbelVandenBriel) [inline packages/semcom-node/lib/store/services/component-pod-store.service.spec.ts:158]: If you use these components multiple times, it's better to set them in a constant at the top of the file, and then set them where you need them.

Would become: `const componentInfos = [ component1, component2 ]` for example

File: packages/semcom-node/lib/store/services/component-pod-store.service.spec.ts [hunk 5]
```diff
@@ -0,0 +1,215 @@
...
+          label: 'test component',
+          description: 'abc',
+          author: '',
+          tag: 'test',
+          version: '0.123',
+          latest: true,
+          shapes: [ 'http://digita.ai/voc/test#test' ],
+        },
+        filename: 'test1.ttl',
+      };
```
Comment (AbelVandenBriel) [inline packages/semcom-node/lib/store/services/component-pod-store.service.spec.ts:123]: set this in a constant

File: packages/semcom-node/lib/store/services/component-pod-store.service.spec.ts (hunk 4 above)
Comment (AbelVandenBriel) [reply packages/semcom-node/lib/store/services/component-pod-store.service.spec.ts:158]: If you need to make changes to the components, you can also declare them globally with let, define them in a beforeEach, and then change only what needs to be changed in the individual test.

File: packages/semcom-node/lib/store/services/component-pod-store.service.spec.ts [hunk 6]
```diff
@@ -0,0 +1,215 @@
...
+          label: 'test2',
+          description: 'defg',
+          author: '',
+          tag: 'test2',
+          version: '0.123',
+          latest: true,
+          shapes: [ 'http://digita.ai/voc/test#test2' ],
+        },
+        filename: 'test2.ttl',
+      } ];
```
Comment (AbelVandenBriel) [inline packages/semcom-node/lib/store/services/component-pod-store.service.spec.ts:195]: Same comment as other components: create constants, or define them in beforeEach and change what needs to be changed in the test.

File: packages/semcom-node/lib/store/services/pod-sync.service.spec.ts [hunk 7]
```diff
@@ -0,0 +1,348 @@
...
+    httpStatus: number;
+    contents: string;
+    filename: string;
+  }
+
+  const streamToBuffer = (stream: NodeJS.ReadableStream): Promise<string> => new Promise((resolve) => {
+
+    const bufs = [];
+    stream.on('data', (chunk) => bufs.push(chunk));
+    stream.on('end', () => resolve(bufs.join('')));
```
Comment (AbelVandenBriel) [inline packages/semcom-node/lib/store/services/pod-sync.service.spec.ts:47]:
```suggestion
    stream.on('end', () => resolve(Buffer.concat(buf))));
```

File: packages/semcom-node/lib/store/services/pod-sync.service.spec.ts [hunk 8]
```diff
@@ -0,0 +1,348 @@
...
+
+  });
+
+  const mockPods = (localComponentFiles: string[], ... otherPods: RemotePodMock[]) => {
+
+    store.set('storage', otherPods.map((pod) => pod.uri));
+
+    localComponents = new Set(localComponentFiles);
+
+    fetchMock = (fetch as jest.MockedFunction<typeof fetch>);
```
Comment (AbelVandenBriel) [inline packages/semcom-node/lib/store/services/pod-sync.service.spec.ts:57]: Same comment as other file with fetch-mock.

File: packages/semcom-node/lib/store/services/pod-sync.service.spec.ts [hunk 9]
```diff
@@ -0,0 +1,348 @@
...
+          components: [],
+          isOnline: true,
+          httpStatus: 200,
+        },
+        {
+          uri: 'test2.com/',
+          components: [],
+          isOnline: true,
+          httpStatus: 200,
+        },
```
Comment (AbelVandenBriel) [inline packages/semcom-node/lib/store/services/pod-sync.service.spec.ts:185]: set individual test pods as constants or declare them with let and define them in before each if you need to change properties.

Comment (AbelVandenBriel) [reply packages/semcom-node/lib/store/services/pod-sync.service.spec.ts:185]: I won't add this comment to every test here, it's becoming a bit redundant :)
Change it where you see it.

**Review by AbelVandenBriel — CHANGES_REQUESTED** [review]

File: packages/semcom-node/lib/store/services/component-pod-store.service.spec.ts (hunk 2 above)
Comment (jonasbovyn) [reply packages/semcom-node/lib/store/services/component-pod-store.service.spec.ts]: I meant to have the method be called streamToString - i changed it

**Review by AbelVandenBriel — APPROVED** [review]

**Pull request merged**